### PR TITLE
Fixes Near-Instant Confirmations, For Good This Time

### PR DESCRIPTION
**Pull request by kalverra**
## Breaking Changes

Subscriptions are now modified to only deal with headers. Before we were pulling the whole block, which was largely unnecessary and ended up causing errors on chains like Arbitrum. This will require some changes on the tests in core repo, but they're small and give us the benefit of slightly less resource usage.

## Goodies

Performance is better for near-instant confirmations by a little bit, it's cleaner, and compatible with wack chains like Arbitrum.

### CODE REVIEW

**Review by tateexon — APPROVED** [review]
lgtm

File: blockchain/transaction_confirmers.go [hunk 1]
```diff
@@ -49,15 +48,15 @@ func NewTransactionConfirmer(client EVMClient, tx *types.Transaction, minConfirm
 
 // ReceiveBlock the implementation of the HeaderEventSubscription that receives each block and checks
 // tx confirmation
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:50]: Change the comment?

File: blockchain/transaction_confirmers.go [hunk 2]
```diff
@@ -49,15 +48,15 @@ func NewTransactionConfirmer(client EVMClient, tx *types.Transaction, minConfirm
...
-func (t *TransactionConfirmer) ReceiveBlock(block NodeBlock) error {
-	if block.NumberU64() <= t.lastReceivedBlockNum {
+func (t *TransactionConfirmer) ReceiveHeader(header NodeHeader) error {
+	if header.Number.Uint64() <= t.lastReceivedBlockNum {
 		return nil // Block with same number mined, disregard for confirming
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:53]: comment

File: blockchain/transaction_confirmers.go [hunk 3]
```diff
@@ -49,15 +48,15 @@ func NewTransactionConfirmer(client EVMClient, tx *types.Transaction, minConfirm
...
 	}
-	t.lastReceivedBlockNum = block.NumberU64()
+	t.lastReceivedBlockNum = header.Number.Uint64()
 	confirmationLog := log.Debug().
 		Str("Network Name", t.networkConfig.Name).
-		Str("Block Hash", block.Hash().Hex()).
-		Str("Block Number", block.Number().String()).
+		Str("Block Hash", header.Hash().Hex()).
+		Str("Block Number", header.Number.String()).
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:59]: the keys

File: blockchain/transaction_confirmers.go [hunk 4]
```diff
@@ -96,67 +95,82 @@ func (t *TransactionConfirmer) Complete() bool {
...
+		completeChan: make(chan struct{}, 1),
+		context:      ctx,
+		cancel:       ctxCancel,
+		// For events
+		confirmedChan: confirmedChan,
+		errorChan:     errorChan,
 	}
 }
 
 // ReceiveBlock does a quick check on if the tx is confirmed already
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:131]: comment

**Review by gheorghestrimtu — APPROVED** [review]

File: blockchain/transaction_confirmers.go (hunk 4 above)
Comment (kalverra) [reply blockchain/transaction_confirmers.go]: Fixed em all, ty!

**Review by AnieeG — APPROVED** [review]
LGTM

**Pull request merged**